lib/ConnectionPool.cc

```diff
@@ -34,13 +34,13 @@ DECLARE_LOG_OBJECT()
 namespace pulsar {
 
 ConnectionPool::ConnectionPool(const ClientConfiguration& conf, ExecutorServiceProviderPtr executorProvider,
-                               const AuthenticationPtr& authentication, bool poolConnections,
-                               const std::string& clientVersion)
+                               const AuthenticationPtr& authentication, const std::string& clientVersion)
     : clientConfiguration_(conf),
       executorProvider_(executorProvider),
       authentication_(authentication),
-      poolConnections_(poolConnections),
-      clientVersion_(clientVersion) {}
+      clientVersion_(clientVersion),
+      randomDistribution_(0, conf.getConnectionsPerBroker()),
```

Comment:
```suggestion
      randomDistribution_(0, conf.getConnectionsPerBroker() + 1),
```

The `uniform_int_distribution` range is left-closed and right-open

Reply:
From the [doc](https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution), the range is left-closed and right-closed: [a,b].

Reply:
My fault. It's true that `uniform_int_distribution` uses right-closed range.

I wanted to say we should use a left-closed and right-open range: `[0, connectionsPerBroker)` so the initialization of `uniform_int_distribution` should be:

```c++
randomDistribution_(0, conf.getConnectionsPerBroker() - 1),
```

@RobertIndie